src/luascript.cpp

```diff
@@ -12216,16 +12218,25 @@ int LuaScriptInterface::luaItemTypeIsStoreItem(lua_State* L)
 int LuaScriptInterface::luaCombatCreate(lua_State* L)
 {
 	// Combat()
-	pushUserdata<Combat>(L, g_luaEnvironment.createCombatObject(getScriptEnv()->getScriptInterface()));
+	pushSharedPtr(L, g_luaEnvironment.createCombatObject(getScriptEnv()->getScriptInterface()));
 	setMetatable(L, -1, "Combat");
 	return 1;
 }
 
+int LuaScriptInterface::luaCombatDelete(lua_State* L)
+{
+	Combat_ptr* combatPtr = getSharedPtr<Combat_ptr>(L, 1);
+	if (combatPtr) {
+		combatPtr->reset();
+	}
+	return 0;
+}
+
 int LuaScriptInterface::luaCombatSetParameter(lua_State* L)
 {
 	// combat:setParameter(key, value)
-	Combat* combat = getUserdata<Combat>(L, 1);
-	if (!combat) {
+	const Combat_ptr* combatPtr = getSharedPtr<Combat_ptr>(L, 1);
```

Comment:
If you rename it back to `combat`, you don't have to change a single line in the rest of the function :wink: